hack/update-e2e-test.sh.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+filepathandname="test/bats/vault.bats"
+
+sed -i -z "s/kubectl create ns vault/oc delete project vault --ignore-not-found=true\n  kubectl create ns vault\n  kubectl label ns vault security.openshift.io\/scc.podSecurityLabelSync=false pod-security.kubernetes.io\/enforce=privileged pod-security.kubernetes.io\/audit=privileged pod-security.kubernetes.io\/warn=privileged --overwrite\n  oc adm policy add-scc-to-user privileged system:serviceaccount:vault:vault\n  oc adm policy add-scc-to-user privileged system:serviceaccount:vault:vault-csi-provider/g" $filepathandname
```

Comment:
I this this is very error prone. Upstream can update the test scripts at any time and the will be no warning that this `sed` did not match anything, leaving the original content there. CI will fail in an odd way ("we already fixed this before !?")

Is it possible to prepare the namespaces in a way that we can run original vault.bats, but it already finds namespaces + SCCs prepared?

Reply:
To provide a bigger picture. My goal is to have as little work with the CSI driver rebase as possible. Someone who does not know much about the CSI driver can list what commits do we carry (= this PR) and without much thinking is able to rebase it on top of a new CSI driver release.

Ideal case: we introduce a new `openshift-hack/test.sh` script that can work even when `test/bats/vault.bats` changes. For example, the script prepares namespaces + their service accounts, so `vault.bats` already has necessary permissions. Propagating some changes upstream (e.g. run the CSI driver as privileged) might be a good option.

Not so ideal, but usable case: change `vault.bats` directly. During the next rebase our carry patch either applies without changes (= `vault.bats` is the same / similar enough as before) or there is a conflict and the person that does the rebase needs to solve it, i.e. update `vault.bats` with SCCs. At least we know there was a change.

Bad case: `update-e2e-test.sh.sh` introduced in this PR does not apply some `sed` replacements, because `vault.bats` changed. The person who does the rebase does not see that and CI starts failing from mysterious reasons. It may take some time for the person to realize what's wrong.

Reply:
This implies that "_Not so ideal, but usable case_" is the right way to proceed, at least for now. IMO carrying a patch for vault.bats is better than carrying fragile sed expressions that don't tell anything when they fail.

Reply:
Hi @jsafrane 
Thanks for inputs. Fixed as per suggestions. PTAL when ever get a chance.